osu.Game/Rulesets/UI/ModIcon.cs

```diff
@@ -96,6 +96,16 @@ public ModIcon(IMod mod, bool showTooltip = true)
                     Icon = FontAwesome.Solid.Question
                 },
             };
+
+            Action = () =>
+            {
+                if (mod is ICanBeToggledDuringReplay dmod)
+                {
+                    dmod.OnToggle();
+
+                    Colour = dmod.IsDisable ? OsuColour.Gray(0.7f) : Colour = Color4.White;
+                }
+            };
```

Comment:
This cannot be done in `ModIcon`. The class is not only used in the replay player. Behaviours specific to a single screen cannot and should not be added to globally-used classes, it's just bad layering.